pkg/volume/volume.go

```diff
@@ -131,10 +131,8 @@ type Deleter interface {
 
 // Attacher can attach a volume to a node.
 type Attacher interface {
-	Volume
-
 	// Attach the volume specified by the given spec to the given host
-	Attach(spec *Spec, hostName string) error
+	Attach(host VolumeHost, spec *Spec, hostName string) error
```

Comment:
Volume host is created, initialized, and stored on plugin creation in the plugin object. It would be really messy to then have callers expect to pass it back in (an object the plugin knows about at volume creation). Since the `NewAttacher` and `NewDetacher` methods can pull the `host` from the `gcePersistentDiskPlugin`:
```GO
func (plugin *gcePersistentDiskPlugin) NewAttacher() (volume.Attacher, error) {}
func (plugin *gcePersistentDiskPlugin) NewDetacher() (volume.Detacher, error) {}
```
where `gcePersistentDiskPlugin` is
```GO
 type gcePersistentDiskPlugin struct {
 	host volume.VolumeHost		 	host volume.VolumeHost
 }
```
I suggest, removing the host from these method params, and adding the `host` to the new `gcePersistentDiskAttacher` and `gcePersistentDiskDetacher` structs in PR #21709 and having `NewAttacher()` and `NewDetacher()` set them accordingly.